### PR TITLE
Add validation on commands for ids and pubkeys

### LINKED ISSUE #499: Add validation on vega IDs / pubkey
These must be on the following format:
- 64 characters
- all hex valid characters

The validation is to be added on all vega IDs (party, assets, markets, etc) and public keys on all commands.

### PR DESCRIPTION
**Pull request by jeremyletang**
close #499

### CODE REVIEW

**Review by pscott31 — APPROVED** [review]

File: commands/announce_node.go [hunk 1]
```diff
@@ -19,11 +17,14 @@ func checkAnnounceNode(cmd *commandspb.AnnounceNode) Errors {
...
-			errs.AddForProperty("announce_node.vega_pub_key", ErrShouldBeHexEncoded)
-		}
+	} else if !IsVegaPubkey(cmd.VegaPubKey) {
+		errs.AddForProperty("announce_node.vega_pub_key", ErrShouldBeAValidVegaPubkey)
+	}
+
+	if len(cmd.Id) == 0 {
+		errs.AddForProperty("announce_node.id", ErrIsRequired)
+	} else if !IsVegaPubkey(cmd.Id) {
+		errs.AddForProperty("announce_node.id", ErrShouldBeAValidVegaPubkey)
```
Comment (AnExsomnis) [inline commands/announce_node.go:27]: I would make these as separate `if` s as the conditions in order_cancelation.go file like:
`if len(cmd.Id) > 0 && !IsVegaPubKey ... ` for consistency (and looks better IMO).
It works that way also though

File: commands/transaction.go [hunk 2]
```diff
@@ -59,6 +59,8 @@ func CheckTransaction(tx *commandspb.Transaction) (*commandspb.InputData, error)
 		errs.AddForProperty("tx.from", ErrIsRequired)
 	} else if len(tx.GetPubKey()) == 0 {
 		errs.AddForProperty("tx.from.pub_key", ErrIsRequired)
+	} else if !IsVegaPubkey(tx.GetPubKey()) {
+		errs.AddForProperty("tx.from.pub_key", ErrShouldBeAValidVegaPubkey)
```
Comment (AnExsomnis) [inline commands/transaction.go:63]: Same here and below

**Review by AnExsomnis — COMMENTED** [review]
Left some comments for a change for the purpose of consistency in conditions checks.

File: commands/announce_node.go (hunk 1 above)
Comment (jeremyletang) [reply commands/announce_node.go:27]: I've made it this way just so we can get more "accurate" error really. I don't mind merging them though. any thoughts?

**Pull request merged**